cosmos/dbt/graph.py

```diff
@@ -143,33 +144,39 @@ def load_via_dbt_ls(self) -> None:
         if self.select:
             command.extend(["--select", *self.select])
 
-        with self.profile_config.ensure_profile() as (profile_path, env_vars):
-            command.extend(
-                [
-                    "--profiles-dir",
-                    str(profile_path.parent),
-                    "--profile",
-                    self.profile_config.profile_name,
-                    "--target",
-                    self.profile_config.target_name,
-                ]
-            )
+        with tempfile.TemporaryDirectory() as tmpdir:
+            # when we drop support to Python 3.7, we can use:
+            # shutil.copytree(self.project.dir, tmpdir, dirs_exist_ok=True)
+            cwd = Path(tmpdir) / self.project.name
+            shutil.copytree(self.project.dir, cwd)
```

Comment:
it's been raised elsewhere that copying the entire directory isn't a great idea with larger projects. curious why we don't just use the log/target path args?

https://docs.getdbt.com/reference/project-configs/log-path#configuration
https://docs.getdbt.com/reference/project-configs/target-path

Reply:
That's a fair point, I'll change to pass the arguments using the value of `DBT_LOG_PATH`  and `DBT_TARGET_PATH`, if they weren't set, and setting those to the temp dir otherwise

Reply:
Unfortunately, neither the flags nor the env vars necessarily work with versions before dbt 1.6 😢 as discussed here:
https://github.com/astronomer/astronomer-cosmos/issues/411#issuecomment-1668259407

However, we've changed the approach and we are no longer copying the project dir 🎉